src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -106,9 +122,21 @@ export function createReactNodeViewConstructor(
 
     const { dom, contentDOM, component: ReactComponent } = reactNodeView;
 
-    const ContentDOMElementType = contentDOM?.tagName.toLocaleLowerCase() as
-      | keyof ReactHTML
-      | undefined;
+    // Use a span if the provided contentDOM is in the "phrasing" content
+    // category. Otherwise use a div. This is our best attempt at not
+    // breaking the intended content model, for now.
+    //
+    // https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content
+    const ContentDOMWrapper =
+      contentDOM &&
+      (phrasingContentTags.includes(contentDOM.tagName.toLocaleLowerCase())
+        ? "span"
+        : "div");
```

Comment:
We can maybe just yank this out and land it as a simple PR first.

Reply:
👍 #29